docs/content/concepts/cryptography/nautilus/using-nautilus.mdx

```diff
@@ -0,0 +1,338 @@
+
+This design allows the admin to run multiple instances of the same enclave with different public keys, where `config_version` is set to the latest version when creating an `Enclave` object. The admin can register or destroy their `Enclave` objects. 
+
+### Update PCRs
+
+The deployer of the smart contract holds the `EnclaveCap`, which allows for updating the PCRs and enclave public key if the Nautilus server code has been modified. You can retrieve the new PCRs using `make && cat out/nitro.pcrs`. To update the PCRs or register the enclave again, reuse the steps outlined in the previous section.
+
+## Using the verified computation in Move
+
+You can now write your frontend code to interact with the enclave for computation, and then send the resulting data to the Move contract for use. For the weather example, you can request the enclave to retrieve weather data for a specific location:
+
+```shell
+curl -H 'Content-Type: application/json' -d '{"payload": { "location": "San Francisco"}}' -X POST http://<PUBLIC_IP>:3000/process_data
+
+
+{"response":{"intent":0,"timestamp_ms":1744683300000,"data":{"location":"San Francisco","temperature":13}},"signature":"77b6d8be225440d00f3d6eb52e91076a8927cebfb520e58c19daf31ecf06b3798ec3d3ce9630a9eceee46d24f057794a60dd781657cb06d952269cfc5ae19500"}
+```
+
+Then use the values from the enclave response - signature, timestamp, location, and temperature - to call `update_weather` in the Move contract. In this example, the call is demonstrated using a script, but it should be integrated into your dApp frontend.
+
+```shell
+sh ../../update_weather.sh \
+    $EXAMPLES_PACKAGE_ID \
+    $MODULE_NAME \
+    $OTW_NAME \
+    $ENCLAVE_OBJECT_ID \
+    "77b6d8be225440d00f3d6eb52e91076a8927cebfb520e58c19daf31ecf06b3798ec3d3ce9630a9eceee46d24f057794a60dd781657cb06d952269cfc5ae19500" \
+    1744683300000 \
+    "San Francisco" \
+    13
+```
+
+An example of a [created weather NFT](https://testnet.suivision.xyz/object/0xa78e166630c0ed004b3115b474fed15d71f27fc80b68e37d451494c6e815931e) is available on network scanners.
+
+### Signing payload
+
+Signing payloads in Move are constructed using Binary Canonical Serialization (BCS). These must match the structure specified in the enclave’s Rust code when generating the signature; otherwise, signature verification in `enclave.move` might fail.
+
+Write unit tests in both Move and Rust to ensure consistency. See `test_serde()` in `src/nautilus-server/src/app.rs` and the examples in `move/enclave/enclave.move`.
+
+## FAQs
+
+1. There are many TEE providers available. Why did Sui choose AWS Nitro Enclaves initially?
```

Comment:
This was answered on another page. Would be better to create a snippet that can be used in both places using {@include}. Would also suggest not using FAQ and just making these their own section.

Reply:
This is also something i can update.

Reply:
Yeah makes sense to me. Thanks for offering to update - would appreciate it :)

Reply:
Wasn't as clean as I assumed. Kept relatively as is but now have the questions appear on right nav for better discoverability.